### PR TITLE
Another take @ simplifying scio context execution

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 1]
```diff
@@ -411,8 +403,8 @@ object ScioContext {
     new ScioContext(options, artifacts)
 
   /** Create a new [[ScioContext]] instance. */
-  def apply(options: PipelineOptions, runner: ScioRunner): ScioContext =
-    new ScioContext(options, Nil, runner)
+  def apply(options: PipelineOptions, executor: ScioExecutor): ScioContext =
```
Comment (anish749) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: I now noticed that this constructor is public.
Do you think this should also be propagated to `ContextAndArgs` as well?
So we can do something like 
```
val (sc: ScioContext, args: Args) = ContextAndArgs(cmdlineArgs, executor)
```

**Comment by jto** [comment]
We're starting to have a lot of breaking changes in `0.8`. 
Can you write add Scalafix to this PR ?

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala (hunk 1 above)
Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: @anish749 tbh i think for the time I'll keep it only on `ScioContext` still a bit unsure about it's full usefulness

**Comment by regadas** [comment]
Closing in favor of #2092

**Pull request closed without merging**